EngineLayer/Silac/SilacConversions.cs

```diff
@@ -332,267 +269,602 @@ public static SpectraFileInfo GetHeavyFileInfo(SpectraFileInfo originalFile, Sil
+                        {
+                            foreach (SpectraFileInfo info in spectraFileInfo)
+                            {
+                                tempFileToRecycleDictionary[info].Add(peptide.GetIntensity(info));
+                            }
+                        }
+
+                        //aggregate into LL, HL, and HH if they contain 2+ missed cleavages
+                        foreach (SpectraFileInfo info in spectraFileInfo)
                         {
-                            foreach (var id in peak.Identifications)
+                            List<double> values = tempFileToRecycleDictionary[info];
+                            if (values.Count > 3)
                             {
-                                SilacLabel label = GetRelevantLabelFromBaseSequence(id.BaseSequence, silacLabels);
-                                HashSet<FlashLFQ.ProteinGroup> originalGroups = id.proteinGroups;
-                                List<FlashLFQ.ProteinGroup> updatedGroups = new List<FlashLFQ.ProteinGroup>();
-                                foreach (FlashLFQ.ProteinGroup group in originalGroups)
+                                if (values.Count == 6)
                                 {
-                                    string groupName = group.ProteinGroupName;
-                                    if (label == null) //if light
-                                    {
-                                        updatedGroups.Add(group);
-                                    }
-                                    else
-                                    {
-                                        string labelString = "(" + label.OriginalAminoAcid + label.MassDifference;
-                                        int labelIndex = groupName.IndexOf(labelString);
-                                        if (labelIndex != -1) //labelIndex == 1 if a) 2+ peptides are required per protein or b) somebody broke parsimony
-                                        {
-                                            groupName = groupName.Substring(0, labelIndex);
-                                            updatedGroups.Add(new FlashLFQ.ProteinGroup(groupName, group.GeneName, group.Organism));
-                                        }
-                                    }
+                                    //LLLLL, LLLLH, LLLHH, LLHHH, LHHHH, HHHHHH have n combinations of 1, 5, 10, 10, 5, 1
+                                    double LLLL = values[0] + values[1] / 5;
+                                    double LLLH = values[1] * 4 / 5 + values[2] * 4 / 10;
+                                    double LLHH = values[2] * 6 / 10 + values[3] * 6 / 10;
+                                    double LHHH = values[3] * 4 / 10 + values[4] * 4 / 5;
+                                    double HHHH = values[4] / 5 + values[5];
+                                    values = new List<double> { LLLL, LLLH, LLHH, LHHH, HHHH };
```

Comment:
i don't see any unit tests that hit mixed L and H. Only fully labeled is tested.

Reply:
Added (I think?)